### PR TITLE
feat: add ability to favorite contacts

### LINKED ISSUE #1594: Add contacts to favorites
We should add the ability of adding some contacts to a favorite list.
These favorites could be displayed in top of the dashboard, with an easy access.

### PR DESCRIPTION
**Pull request by djaiss**
This PR adds the ability to mark a contact as a favorite. This close #1594
This will put the contact on top of the contact list, as well as appear in a new section on the dashboard.

screen shot 2018-08-15 at 10 52 16 pm

#### Code-related tasks
- [x] Tests added for this feature/bug.
- [x] Impact on the seeders.
- [x] Impact on the API.

#### If the code changes the SQL schema
- [x] Impact on account export.
- [x] Impact on importing data with `vCard` and `.csv` files.
- [x] Impact on account reset and deletion.

#### Other tasks
- [x] [CHANGELOG](https://github.com/monicahq/monica/blob/master/CHANGELOG) entry added, if necessary, under `UNRELEASED`.

### CODE REVIEW

File: resources/assets/js/components/people/SetFavorite.vue [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<style scoped>
+</style>
+
+<template>
+    <div>
+        <notifications group="favorite" position="top middle" duration="5000" width="400" />
+        <p>
+            <a class="pointer" @click="store(!isFavorite)" v-if="!isFavorite">{{ $t('people.set_favorite') }}</a>
```
Comment (asbiin) [inline resources/assets/js/components/people/SetFavorite.vue]: You could call `store(true)` here

File: resources/assets/js/components/people/SetFavorite.vue [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            <a class="pointer" @click="store(!isFavorite)" v-if="isFavorite">{{ $t('people.unset_favorite') }}</a>
```
Comment (asbiin) [inline resources/assets/js/components/people/SetFavorite.vue]: You could call `store(false)` here

**Comment by asbiin** [comment]
I'm not sure about the texts 'Set as favorite' and 'Remove favorite'. What about 'Add to favorites' and 'Remove from favorites'?

**Comment by djaiss** [comment]
@asbiin ready for final review

File: resources/assets/js/components/people/SetFavorite.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        </svg>
+
+        <svg class="pointer" @click="store(false)" :cy-name="'unset-favorite'" v-if="isFavorite" width="23" height="22" viewBox="0 0 23 22" fill="none" xmlns="http://www.w3.org/2000/svg">
```
[the remark this answers is not in the document]
Comment (djaiss) [reply resources/assets/js/components/people/SetFavorite.vue]: I did it on purpose as you've already set someone as favorite and you don't need to have this information to "unset" the favorite. What do you think?

[the remark this answers is not in the document]
Comment (djaiss) [reply resources/assets/js/components/people/SetFavorite.vue]: When I don't set the contact as favorite: i tell the user what the feature does.
When the contact is a favorite: I don't need to tell something?

[the remark this answers is not in the document]
Comment (asbiin) [reply resources/assets/js/components/people/SetFavorite.vue]: But I trust your opinion, so be it (and we could always improve it later).

**Pull request merged**